File: HomeWork/Home11.1/BookRecommender.java
import java.util.ArrayList;
import java.util.List;
import java.util.Random;
import java.util.Scanner;

public class BookRecommender {
    private List<String> books;

    public BookRecommender() {
        books = new ArrayList<>();
        books.add("The Topographer’s Clown");
        books.add("The Chamber of Beaver");
        books.add("The Ironer of Kanban");
        books.add("The Piglet of Tire");
        books.add("The Border of the Unix");
        books.add("The Half-Time Convince");
        books.add("The Earthly Pillows");
        books.add("The Censorship of the Ping");
        books.add("The True Powers");
        books.add("The Overturn of the Ling");
    }

    private String getRandomBook() {
        if (books.isEmpty()) {
            return "";
        }
        Random rand = new Random();
        int index = rand.nextInt(books.size());
        return books.get(index);
    }

    public void recommendBook(String name) {
        Scanner scanner = new Scanner(System.in);
        while (!books.isEmpty()) {
            String book = getRandomBook();
            System.out.println("We recommend you to read: " + book);
            System.out.print("Do you like this book? (yes/no): ");
            String response = scanner.nextLine();

            if (response.equalsIgnoreCase("yes")) {
                System.out.println("Great! Enjoy your book.");
                return;
            } else if (response.equalsIgnoreCase("no")) {
                books.remove(book);
            } else {
                System.out.println("Invalid response. Please enter 'yes' or 'no'.");
            }
        }
        System.out.println("Au revoir, " + name + "!");
        scanner.close();
    }
}

File: Class/practice11.1/BookRecommender.cpp
#include "BookRecommender.h"
#include <iostream>
#include <algorithm>
#include <cstdlib>
#include <ctime>

BookRecommender::BookRecommender() {
    books = {
        "The Topographer’s Clown",
        "The Chamber of Beaver",
        "The Ironer of Kanban",
        "The Piglet of Tire",
        "The Border of the Unix",
        "The Half-Time Convince",
        "The Earthly Pillows",
        "The Censorship of the Ping",
        "The True Powers",
        "The Overturn of the Ling"
    };
    std::srand(std::time(0));  // Seed random number generator
}

std::string BookRecommender::getRandomBook() {
    if (books.empty()) {
        return "";
    }
    int index = std::rand() % books.size();
    return books[index];
}

void BookRecommender::recommendBook() {
    while (!books.empty()) {
        std::string book = getRandomBook();
        std::cout << "We recommend you to read: " << book << std::endl;
        std::cout << "Do you like this book? (yes/no): ";
        std::string response;
        std::cin >> response;

        if (response == "yes") {
            std::cout << "Great! Enjoy your book." << std::endl;
            return;
        } else if (response == "no") {
            books.erase(std::remove(books.begin(), books.end(), book), books.end());
        } else {
            std::cout << "Invalid response. Please enter 'yes' or 'no'." << std::endl;
        }
    }
    std::cout << "Adios amigo!" << std::endl;
}
